src/Identity/Controllers/AccountsController.cs

```diff
@@ -18,27 +19,32 @@ public class AccountsController : Controller
     private readonly ILogger<AccountsController> _logger;
     private readonly IUserRepository _userRepository;
     private readonly IUserService _userService;
+    private readonly ICaptchaValidationService _captchaValidationService;
 
     public AccountsController(
         ILogger<AccountsController> logger,
         IUserRepository userRepository,
-        IUserService userService)
+        IUserService userService,
+        ICaptchaValidationService captchaValidationService)
     {
         _logger = logger;
         _userRepository = userRepository;
         _userService = userService;
+        _captchaValidationService = captchaValidationService;
     }
 
     // Moved from API, If you modify this endpoint, please update API as well.
     [HttpPost("register")]
     [CaptchaProtected]
-    public async Task PostRegister([FromBody] RegisterRequestModel model)
+    public async Task<RegisterResponseModel> PostRegister([FromBody] RegisterRequestModel model)
     {
         var result = await _userService.RegisterUserAsync(model.ToUser(), model.MasterPasswordHash,
             model.Token, model.OrganizationUserId);
         if (result.Succeeded)
         {
-            return;
+            var user = await _userRepository.GetByEmailAsync(model.Email);
```

Comment:
You don't need to grab this user again. All you need to do is store the `model.ToUser` and it'll be updated in place.

```csharp
var user = model.ToUser();
```

then you can pass that `user` directly to captchaValudationService.

Reply:
Done!